### PR TITLE
Fix NPEs during realm import

### LINKED ISSUE #10961: Various NPEs using the realm create REST API
### Describe the bug

Trying to create a new realm using the REST API throws various NPEs instead of gracefully handling the situation.

This is the stacktrace of the first and outstanding error:

```
2022-03-28 10:25:02,036 ERROR [org.keycloak.services.error.KeycloakErrorHandler] (executor-thread-1) Uncaught server error: java.lang.NullPointerException
        at org.keycloak.models.utils.RepresentationToModel.importAuthenticationFlows(RepresentationToModel.java:732)
        at org.keycloak.models.utils.RepresentationToModel.importRealm(RepresentationToModel.java:306)
        at org.keycloak.services.managers.RealmManager.importRealm(RealmManager.java:557)
        at org.keycloak.services.managers.RealmManager.importRealm(RealmManager.java:495)
        at org.keycloak.services.resources.admin.RealmsAdminResource.importRealm(RealmsAdminResource.java:133)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:170)
        at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:130)
        at org.jboss.resteasy.core.ResourceMethodInvoker.internalInvokeOnTarget(ResourceMethodInvoker.java:660)
        at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTargetAfterFilter(ResourceMethodInvoker.java:524)
        at org.jboss.resteasy.core.ResourceMethodInvoker.lambda$invokeOnTarget$2(ResourceMethodInvoker.java:474)
        at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:364)
        at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTarget(ResourceMethodInvoker.java:476)
        at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:434)
        at org.jboss.resteasy.core.ResourceLocatorInvoker.invokeOnTargetObject(ResourceLocatorInvoker.java:192)
        at org.jboss.resteasy.core.ResourceLocatorInvoker.invoke(ResourceLocatorInvoker.java:141)
        at org.jboss.resteasy.core.ResourceLocatorInvoker.invoke(ResourceLocatorInvoker.java:32)
        at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:492)
        at org.jboss.resteasy.core.SynchronousDispatcher.lambda$invoke$4(SynchronousDispatcher.java:261)
        at org.jboss.resteasy.core.SynchronousDispatcher.lambda$preprocess$0(SynchronousDispatcher.java:161)
        at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:364)
        at org.jboss.resteasy.core.SynchronousDispatcher.preprocess(SynchronousDispatcher.java:164)
        at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:247)
        at io.quarkus.resteasy.runtime.standalone.RequestDispatcher.service(RequestDispatcher.java:73)
        at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.dispatch(VertxRequestHandler.java:151)
        at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.handle(VertxRequestHandler.java:82)
        at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.handle(VertxRequestHandler.java:42)
        at io.vertx.ext.web.impl.RouteState.handleContext(RouteState.java:1212)
        at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:163)
        at io.vertx.ext.web.impl.RoutingContextImpl.next(RoutingContextImpl.java:141)
        at io.quarkus.vertx.http.runtime.StaticResourcesRecorder$2.handle(StaticResourcesRecorder.java:67)
        at io.quarkus.vertx.http.runtime.StaticResourcesRecorder$2.handle(StaticResourcesRecorder.java:55)
        at io.vertx.ext.web.impl.RouteState.handleContext(RouteState.java:1212)
        at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:163)
        at io.vertx.ext.web.impl.RoutingContextImpl.next(RoutingContextImpl.java:141)
        at io.quarkus.vertx.http.runtime.VertxHttpRecorder$5.handle(VertxHttpRecorder.java:380)
        at io.quarkus.vertx.http.runtime.VertxHttpRecorder$5.handle(VertxHttpRecorder.java:358)
        at io.vertx.ext.web.impl.RouteState.handleContext(RouteState.java:1212)
        at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:163)
        at io.vertx.ext.web.impl.RoutingContextImpl.next(RoutingContextImpl.java:141)
        at org.keycloak.quarkus.runtime.integration.web.QuarkusRequestFilter.lambda$createBlockingHandler$1(QuarkusRequestFilter.java:71)
        at io.vertx.core.impl.ContextImpl.lambda$null$0(ContextImpl.java:159)
        at io.vertx.core.impl.AbstractContext.dispatch(AbstractContext.java:100)
        at io.vertx.core.impl.ContextImpl.lambda$executeBlocking$1(ContextImpl.java:157)
        at io.quarkus.vertx.core.runtime.VertxCoreRecorder$13.runWith(VertxCoreRecorder.java:543)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
        at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
        at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:829)
```

### Version

latest

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by andreaTP**
Resolves #10961

### CODE REVIEW

**Review by vmuzikar — CHANGES_REQUESTED** [review]
We should probably add some test for the fixed NPEs.

**Comment by andreaTP** [comment]
@vmuzikar would you like to indicate an already available test that I can expand to cover those cases?

**Comment by vmuzikar** [comment]
@andreaTP Maybe this one? https://github.com/keycloak/keycloak/blob/main/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/realm/RealmTest.java

**Comment by andreaTP** [comment]
@vmuzikar added a test in [ExportImportTest.java](https://github.com/keycloak/keycloak/pull/10962/files#diff-26be205630c83c4e439ca72e46df08135c63aa3c574d11bf0bee91f3e5f2aed1) let see if the CI is happy 🙂

File: testsuite/integration-arquillian/tests/base/src/test/resources/import/testrealm-authenticator-config-null.json [hunk 1]
```diff
@@ -0,0 +1,564 @@
...
+              }
+           ],
+           "builtIn": true,
+           "description": "browser based authentification2",
+           "id": "3e6ccf87-5473-4eb0-8cbb-28f6b9e6f4d6",
+           "providerId": "basic-flow",
+           "topLevel": true
+        }
+     ],
+     "clientScopes": [
```
Comment (mposolda) [inline testsuite/integration-arquillian/tests/base/src/test/resources/import/testrealm-authenticator-config-null.json:21]: Minor: Is it needed that this JSON contains "clientScopes" and 'userFederationMappers" and "userFederationProviders" ? AFAIK not and in this case, I vote for removing the unneeded parts to make it easier for the maintenance in the future.

**Review by mposolda — CHANGES_REQUESTED** [review]
@andreaTP Thanks! PR looks good, I've added only one minor comments regarding removing some uneeded mess from JSON.

File: testsuite/integration-arquillian/tests/base/src/test/resources/import/testrealm-authenticator-config-null.json (hunk 1 above)
Comment (andreaTP) [reply testsuite/integration-arquillian/tests/base/src/test/resources/import/testrealm-authenticator-config-null.json]: removed

**Comment by andreaTP** [comment]
@mposolda addressed your comment and squashed commits. Ready for another check.

**Review by mposolda — APPROVED** [review]
@andreaTP Thanks!

**Review by vmuzikar — APPROVED** [review]

**Comment by mposolda** [comment]
@andreaTP @vmuzikar Thanks for the changes and review!

**Pull request merged**